### PR TITLE
Pwa 2172: Validate Shopping Cart Promotions are displayed correctly

### PR DESCRIPTION
**Pull request by michaelyu0123**
## Description

As a merchant I want to be able to personalize promotions through segmentation so that I can provide the best possible experience to my customers

Acceptance Criteria

Shopping cart price rules should be applied to relevant carts accurately
Cart price rules based on segmentation must reflect the accurate customer segment
Display Shopping Cart Price Rule Label in cart summary

Example Scenarios

Free Shipping > $50 cart total, no exclusions
Free Shipping for Registered Customers only - registered customer segment applied to rule
10% Discount over $500 Cart value for Guest Customers - guest customer segment applied to rule, not applicable for registered users
15% Discount over $500 Cart value for Registered Customers - registered customer segment applied to rule, not applicable for registered users
20% Discount for high spending customers - registered customer segment, total historical order value > $1000

## Related Issue

Closes #[Pwa 2172](https://jira.corp.magento.com/browse/PWA-2172)

## Verification Steps

Test Plan
Execute -

Verify cart price rules are displayed based on how they are configured in backend.
Verify all scenarios from the example.
Verify Merge cart flow from guest to registered user where guest cart has some rule applied and registered user cart has something different.
Test Coverage -

Add cypress integration test coverage.

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/venia-ui/package.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
     "lodash.throttle": "~4.1.1",
     "memoize-one": "~5.0.0",
     "prop-types": "~15.7.2",
+    "react-animate-height": "^2.0.23",
```
Comment (ericeoeur) [inline packages/venia-ui/package.json:44]: Why was this dependency added?

Comment (ericeoeur) [reply packages/venia-ui/package.json:44]: @michaelyu0123, just left that comment above the end of last week with James and it seems that some conflicts just came up that need to be resolved.

**Comment by dpatil-magento** [comment]
@magento give me 2.4.3

File: packages/venia-ui/package.json (hunk 1 above)
Comment (michaelyu0123) [reply packages/venia-ui/package.json:44]: this was added to create smooth dropdown for discount summary

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run cypress

**Comment by michaelyu0123** [comment]
run cypress

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run cypress

**Comment by michaelyu0123** [comment]
run cypress

**Comment by michaelyu0123** [comment]
run all tests

**Comment by michaelyu0123** [comment]
run cypress

**Comment by michaelyu0123** [comment]
run cypress

**Review by ericeoeur — COMMENTED** [review]
✅ Approved, Ready for QA

**Comment by mikhaelbois** [comment]
@michaelyu0123 
- Free Shipping > $50 cart total, no exclusions ✅ 
The free shipping is applied.
Found this issue with the Table Rate that did not show the right price compared to Luma.
But this can be checked in another task.
image
image
- Free Shipping for Registered Customers only ✅
- 10% Discount over $500 Cart value for Guest Customers ✅ 
- 15% Discount over $500 Cart value for Registered Customers ✅
- 20% Discount for high spending customers ✅

**Comment by mikhaelbois** [comment]
run all tests

**Comment by mikhaelbois** [comment]
run cypress

**Review by ericeoeur — COMMENTED** [review]
Approved✅

**Review by mikhaelbois — APPROVED** [review]

**Comment by mikhaelbois** [comment]
run cypress

**Comment by mikhaelbois** [comment]
run all tests

**Pull request merged**